pennylane/devices/qutrit_mixed/initialize_state.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions to prepare a qutrit mixed state."""
+
+from typing import Iterable, Union
+import pennylane as qml
+from pennylane import (
+    QutritBasisState,
+)
+from pennylane.operation import (
+    StatePrepBase,
+    Operation,
+)
+
+qudit_dim = 3  # specifies qudit dimension
+
+
+def create_initial_state(
+    wires: Union[qml.wires.Wires, Iterable],
+    prep_operation: Operation = None,
+    like: str = None,
+):
+    r"""
+    Returns an initial state, defaulting to :math:`\ket{0}\bra{0}` if no state-prep operator is provided.
+
+    Args:
+        wires (Union[Wires, Iterable]): The wires to be present in the initial state
+        prep_operation (Optional[StatePrepBase]): An operation to prepare the initial state
+        like (Optional[str]): The machine learning interface used to create the initial state.
+            Defaults to None
+
+    Returns:
+        array: The initial state of a circuit
+    """
+    num_wires = len(wires)
+
+    if not prep_operation:
+        rho = _create_basis_state(num_wires, 0)
+
+    elif isinstance(prep_operation, QutritBasisState):
+        rho = _apply_basis_state(prep_operation.parameters[0], num_wires)
```

Comment:
I believe that won't be possible as that can be dealt with in preprocessing, line 40 of ```devices/qubit/initialize_state.py``` uses the same method to get the number of wires.

Reply:
I think @albi3ro 's concern is valid. `devices/qubit/initialize_state.py` gets the state vector directly from the state prep operation, which expands the state vector over the specified wire order. That wire order can't be specified here, which will be problematic. For example, if someone wants to create a state with 4 wires, but the `QutritBasisState` operator is defined as `qml.QutritBasisState([2, 0], wires=[0, 1]), the expected state is supposed to be $|2000\rangle$, but will actually be $|0020\rangle$ because you just get the index to be 6, not 54. So this should actually be:

```suggestion
        rho = _apply_state_vector(prep_operation.state_vector(wire_order=wires), num_wires)
```

In other words, all `StatePrepBase` operations, including `QutritBasisState` should dispatch to what you currently have on line 55.

Reply:
Ok, I agree, that makes sense, thank you for catching this @albi3ro!

Reply:
Thank you for helping me understand @mudit2812